File: TarjetaEstilosVueDasboard/src/models/ItemContent.ts
class ItemContent{
    icon:string|null;
    description:string|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    id:number|null;
    page_content_id:number|null

    constructor(
        icon:string|null,
        description:string|null,
        created_at:Date|null, 
        updated_at:Date|null, 
        deleted_at:Date|null, id:number|null=null,
        page_content_id:number|null=null
    ){
        this.icon = icon;
        this.description = description;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.id = id;
        this.page_content_id = page_content_id;
    }
};
export default ItemContent;

File: src/main/java/pe/tarjetaestilos/models/ItemContent.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;
import java.time.ZonedDateTime;
import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name="item_content")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class ItemContent {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(columnDefinition = "TEXT")
    private String icon;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @ManyToOne()
    @JoinColumn(name = "page_content_id", nullable = false)
    @ToString.Exclude
    @JsonBackReference
    private PageContent pageContent;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;
    
    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;
    
}
